partition/src/syscall.rs

```diff
@@ -0,0 +1,157 @@
+//! Implementation of the mechanism to perform system calls
+
+// TODO: Document the mechanism here
+
+use std::io::IoSlice;
+use std::os::fd::{AsRawFd, RawFd};
+
+use a653rs_linux_core::mfd::Mfd;
+use a653rs_linux_core::syscall::{SyscallRequ, SyscallResp};
+use anyhow::Result;
+use nix::libc::EINTR;
+use nix::sys::eventfd::{self, EfdFlags};
+use nix::sys::socket::{sendmsg, ControlMessage, MsgFlags};
+use polling::{Event, Poller};
+
+use crate::SYSCALL;
+
+/// Sends a vector of file descriptors through a Unix socket
+fn send_fds(hv: RawFd, fds: &Vec<RawFd>) -> Result<()> {
```

Comment:
can we just use a slice and not a `&Vec<RawFd>`? That would spare us unnecessary allocations, I think most of the time we know at compiletime how many fds we need, thus an [RawFd; N]` would suffice.

Reply:
Sounds good